clients/CMakeLists.txt

```diff
@@ -22,6 +22,8 @@ else()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp -pthread")
 endif()
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ftemplate-depth=2048")
```

Comment:
Why is this necessary? In general, it isn't good to go beyond the default template depth.

Reply:
> 
> 
> Why is this necessary? In general, it isn't good to go beyond the default template depth.

It was necessary because libstd++ uses an inefficient recursive method for computing `std::is_nothrow_move_constructible`. I don't have references handy, but I heavily researched it. I thought about adding a `noexcept(false)` clause to one level of our `std::tuple` usage, on the chance that it might short-circuit the inefficient `std::is_nothrow_move_constuctible` algorithm, but it seemed like increasing the limit was simpler. We only had to double it, approximately.

We use `std::tuple` with 46 elements. We don't use recursion ourselves, but rather use `std::index_sequence` and the like, i.e., we use iterative rather than recursive algorithms. It's libstdc++ which uses an inefficient recursive algorithm for `std::is_nothrow_move_constructible`.

Reply:
c.f. https://code-examples.net/en/q/164ac69

Reply:
Then maybe you should consider using a lambda instead of tuple. I present one possible solutions.

Reply:
`std::tuple` is rather slow for compile-time and inefficient implementation(like in libstdc++) makes it worse.

Reply:
> 
> 
> `std::tuple` is rather slow for compile-time and inefficient implementation(like in libstdc++) makes it worse.

This is only for our tests, not for the main library, so performance and compile-time efficiency are not the most important concerns (code readability and reusability are more important).

Reply:
You'll notice that the increased limit is only in `clients/CMakeLists.txt`. It is not necessary that it go into `library/CMakeLists.txt`. That's what I mean by saying it's only in the tests, not the library.

Reply:
We still need to use `std::tuple` in some places, but rarely with the full `Arguments` list. With the subsets we use, we no longer need this limit increase.

As I said before, we don't use recursion (we use `std::index_sequence` and the like). The overflow was caused by the inefficient recursive implementation of libstdc++'s `std::is_nothrow_move_constructible`.